File: backend/app/api/v1/schemas/audio.py
# app/api/v1/schemas/audio.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
import uuid


class AudioFileResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    original_filename: str
    format: str
    duration_secs: Optional[float] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True


class AudioListResponse(BaseModel):
    files: list[AudioFileResponse]
    total: int

File: frontend/src/types/audio.ts
/**
 * Types et interfaces relatifs aux fichiers audio.
 */

export type AudioStatus =
  | "uploaded"
  | "converting"
  | "converted"
  | "transcribing"
  | "correcting"
  | "transcribed"
  | "error";

export interface AudioFile {
  id: string;
  project_id: string;
  original_filename: string;
  format: string;
  duration_secs: number | null;
  storage_path_raw?: string;
  storage_path_converted?: string;
  status: AudioStatus;
  created_at: string;
}

export interface AudioListResponse {
  files: AudioFile[];
  total: number;
}
